base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
+
+# temp definitions while Transpose is not subtype of AbstractMatrix
+# Base.eltype{T}(A::Transpose{T}) = T
+
+# end temp
+
+Base.size(A::Transpose) = size(A.data, 2), size(A.data, 1)
+function Base.size(A::Transpose, i::Integer)
+    if i <= 0
+        error("arraysize: dimension out of range")
+    elseif i <= 2
+        return size(A.data, 3 - i)
+    else
+        return 1
+    end
+end
+
+@inline function Base.getindex(A::Transpose, i::Integer, j::Integer)
+    aji = A.data[j,i]
+    return ifelse(A.conjugated, conj(aji), aji)
+end
+
+Base.similar(A::Transpose, T::Type, d::Dims) = similar(A.data, T, d)
+
+ctranspose(A::AbstractMatrix) = Transpose{eltype(A), typeof(A)}(A, true)
+function ctranspose(A::Transpose)
+    if A.conjugated
+        return A.data
+    else
+        return map(conj, A.data)
+    end
+end
```

Comment:
Strange that `ctranspose` sometimes returns a view and sometimes returns a near-copy. Why not clear this as
```jl
ctranspose(A::Transpose) = Transpose(A.data, !A.conjugated)
```
?